src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,18 @@
+import { deleteComment } from "../utils/api";
+
+const DeleteComment = ({ comment_id, setDeleted, setComments }) => {
+  const handleClick = () => {
+    deleteComment(comment_id).then(() => {
+      setDeleted(true);
+      setComments(comments => [...comments]);
+    });
```

Comment:
This achieves the aim of re-fetching the comments once a comment has been deleted, but it is a _little_ hacky. It works because you're setting the comments with an exact copy itself, which your `useEffect` in the parent component considers to be a "change" in the state, and so reruns itself and fetches the comments again. Perhaps instead, inside the `handleClick` function you can just fetch comments once more and `setComments` with that new array. Achieves the same thing but is a bit clearer to follow the line of what's going on!